_signal_processing/krakenSDR_signal_processor.py

```diff
@@ -373,255 +459,295 @@ def run(self):
-                            else:
-                                self.logger.error(f"Invalid DOA Result data format: {self.DOA_data_format}")
-
-                            if self.hasgps and self.usegps:
-                                self.update_location()
-                    except:
+                                    self.theta_0_list.append(theta_0)
+                                    self.confidence_list.append(np.max(conf_val))
+                                    self.max_power_level_list.append(np.maximum(-100, max_amplitude))
+                                    self.freq_list.append(write_freq)
+                                    self.doa_result_log_list.append(doa_result_log)
+
+                            que_data_packet.append(["doa_thetas", self.DOA_theta])
+                            que_data_packet.append(["DoA Result", doa_result_log])
+                            que_data_packet.append(["DoA Max", theta_0])
+                            que_data_packet.append(["DoA Confidence", conf_val])
+                            que_data_packet.append(["DoA Squelch", update_list])
+                            que_data_packet.append(["DoA Max List", self.doa_max_list])
+                    except Exception:
                         self.data_ready = False
 
                     # -----> SPECTRUM PROCESSING <-----
                     if self.en_spectrum and self.data_ready:
-                        spectrum_plot_data = reduce_spectrum(self.spectrum, self.spectrum_plot_size,
-                                                             self.channel_number)
-                        que_data_packet.append(['spectrum', spectrum_plot_data])
+                        spectrum_plot_data = reduce_spectrum(
+                            self.spectrum, self.spectrum_plot_size, self.channel_number
+                        )
+                        que_data_packet.append(["spectrum", spectrum_plot_data])
 
                     # Record IQ samples
                     if self.en_record:
                         # TODO: Implement IQ frame recording
                         self.logger.error(
-                            "Saving IQ samples to npy is obsolete, IQ Frame saving is currently not implemented")
+                            "Saving IQ samples to npy is obsolete, IQ Frame saving is currently not implemented"
+                        )
+
+                    daq_cpi = int(
+                        self.module_receiver.iq_header.cpi_length * 1000 / self.module_receiver.iq_header.sampling_freq
+                    )
+                    # We don't include processing latency here, because reported timestamp marks end of the data frame
+                    # so latency is essentially an acquisition time.
+                    self.latency = daq_cpi
+
+                    if self.data_ready and self.theta_0_list:
+                        # Do Kraken App first as currently its the only one supporting multi-vfo out
+                        if (
```

Comment:
It is strange `if (` formating, it is not python way. Python IDE and extensions could complain on it

Reply:
Closing, because looks like it is okay according the PEP-8